pytest_freezegun.py

```diff
@@ -1,18 +1,52 @@
 # -*- coding: utf-8 -*-
 
+import time
+
+import _pytest.runner
 import pytest
+
 from freezegun import freeze_time
 
 
-@pytest.yield_fixture
+@pytest.fixture
 def freezer(request):
-    marker = request.node.get_marker('freeze_time')
-    args, kwargs = (marker.args, marker.kwargs) if marker else ((), {})
-    with freeze_time(*marker.args, **marker.kwargs) as freezer:
-        yield freezer
+    freezer, frozen_time = FreezegunPlugin.freezers.get(
+        request.node.nodeid,
+        (None, None)
+    )
+
+    if freezer is None:
+        with freeze_time() as frozen_time:
+            yield frozen_time
+    else:
+        yield frozen_time
+
+
+def patch_pytest_runner_time():
+    # Patch the time function in pytest's runner,
+    # so it reports the correct setup/call/teardown durations.
+    _pytest.runner.time = time.time.previous_time_function
```

Comment:
Could this be done by passing `ignore=('_pytest.runner', )` to `freeze_time`? Or maybe even whole `_pytest` - I'll have to check, but I have a feeling junit reporting is broken too.

Reply:
looks like junit is OK - it's using the pytest_sessionstart/pytest_sessionfinish hooks to get the session time, and does not measure test durations by itself.

Reply:
You're absolutely right. I didn't know about `ignore`.